src/libraries/System.Text.Encoding/tests/Ascii/EqualsTests.cs

```diff
@@ -20,15 +20,35 @@ public abstract class AsciiEqualityTests<TLeft, TRight>
         protected abstract bool Equals(ReadOnlySpan<TLeft> left, ReadOnlySpan<TRight> right);
         protected abstract bool EqualsIgnoreCase(ReadOnlySpan<TLeft> left, ReadOnlySpan<TRight> right);
 
+        private static readonly int[] BufferLengths =
+        [
+            1,
+            (sizeof(long) / sizeof(short)) - 1,
+            (sizeof(long) / sizeof(short)),
+            (sizeof(long) / sizeof(short)) + 1,
+            Vector128<short>.Count - 1,
+            Vector128<short>.Count,
+            Vector128<short>.Count + 1,
+            Vector256<short>.Count - 1,
+            Vector256<short>.Count,
+            Vector256<short>.Count + 1,
+            Vector512<short>.Count - 1,
+            Vector512<short>.Count,
+            Vector512<short>.Count + 1
+        ];
+
         public static IEnumerable<object[]> ValidAsciiInputs
         {
             get
             {
                 yield return new object[] { "test" };
 
-                for (char textLength = (char)0; textLength <= 127; textLength++)
+                foreach (int textLength in BufferLengths)
```

Comment:
That is a lot of test cases, I am not sure how long it's going to take to run all of them with debug builds. 

Before your change, this test  was covering strings that were 0 to 127 chars long. Now the max length is reduced to 33. I am not convinced that this is the right thing to do, please revert this particular change or convince me that it's the right thing to do.